src/hooks/useConnectedDelegate.ts

```diff
@@ -1,25 +1,58 @@
-import { fetchConnectedDelegate, } from "@/app/delegates/actions";
+import { fetchConnectedDelegate, revalidateDelegateAddressPage } from "@/app/delegates/actions";
 import { useAccount } from "wagmi";
 import { Delegate } from "@/app/api/common/delegates/delegate";
 import { useState, useCallback, useEffect } from "react";
+import { useConnectButtonContext } from "@/contexts/ConnectButtonContext";
 
-// TODO: think about strategy to refetchConnectedDelegate, since balance and voting power can change on every block
+function timeout(ms: number) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+// TODO: think about strategy to fetchConnectedDelegate, since balance and voting power can change on every block, 
+// also to prevent additional unnecessary fetches being done right now
 const useConnectedDelegate = () => {
+  const { refetchDelegate, setRefetchDelegate } = useConnectButtonContext();
   const { address } = useAccount();
   const [delegate, setDelegate] = useState<Delegate | null>(null);
   const [advancedDelegators, setAdvancedDelegators] = useState<string[] | null>(
     null
   );
   const [balance, setBalance] = useState<bigint | null>(null);
+  const [retries, setRetries] = useState<number>(0);
+  const [lastVotingPower, setLastVotingPower] = useState<string | null>(null);
 
   const fetchDelegateAndSet = useCallback(async (address: string) => {
     if (address) {
       const [delegate, advancedDelegators, balance] = await fetchConnectedDelegate(address);
+      /**
+       * Assuming delegator and delegatee data update at the same time in the db, since we are revalidating 
+       * the delegatee until the delegator voting power is updated in the DB. If delegatee is updated later than 
+       * delegator after a delegation this solution does not work
+       */
```

Comment:
Hi @stepandel I added this comment, since from my understanding delegatee and delegator voting power should be updated at the same time in the DB after a delegation right?